src/dag.ts

```diff
@@ -0,0 +1,85 @@
+import { Model } from './model';
+import { Script } from './script';
+import { Tensor } from './tensor';
+import { Dtype, DTypeMap } from './dtype';
+
+export interface DagCommandInterface {
+  tensorset(keName: string, t: Tensor);
+
+  tensorget(keyName: string);
+
+  tensorget(keyName: string);
+
+  modelrun(modelName: string, inputs: string[], outputs: string[]);
+
+  scriptrun(scriptName: string, functionName: string, inputs: string[], outputs: string[]);
+}
+
+/**
+ * Direct mapping to RedisAI DAGs
+ */
+export class Dag implements DagCommandInterface {
+  private _commands: any[][];
+  private _tensorgetflag: boolean[];
+
+  constructor() {
+    this._commands = [];
+    this._tensorgetflag = [];
+  }
+
+  public tensorset(keName: string, t: Tensor) {
```

Comment:
Can we change to chained calls here? like: `public tensorset(keName: string, t: Tensor) : Dag ` and `ends with return this;`, Not sure if this is supported : )

Reply:
Yes! great idea. Added the changes and an test case with it:
```
dag
      .tensorset('tensorA', tensorA)
      .tensorset('tensorB', tensorB)
      .modelrun('mymodel-dag', ['tensorA', 'tensorB'], ['tensorC'])
      .tensorget('tensorC');
```
Thank you @dengliming =)